### PR TITLE
Simplify parseQueryString

### PR DESCRIPTION
**Pull request by jridgewell**
And it's [faster](https://jsperf.com/parsing-query-string).

### CODE REVIEW

File: src/url-parse-query-string.js [hunk 1]
```diff
@@ -31,25 +33,12 @@ export function parseQueryString_(queryString) {
...
-      value = '';
-    }
-    if (name) {
-      params[name] = value;
-    }
+
+  let match;
+  while ((match = regex.exec(queryString))) {
+    const name = decodeURIComponent(match[1]).trim();
+    const value = match[2] ? decodeURIComponent(match[2]).trim() : '';
```
Comment (dvoytenko) [inline src/url-parse-query-string.js]: Just realized that trimming the result of decode is probably a wrong thing :( Maybe we should do it in a separate PR, but `x=value%20` should probably preserve a space in the value? Trimming pre-decoded strings is certainly fine...

Comment (jridgewell) [inline src/url-parse-query-string.js]: I'm kinda of the opinion we shouldn't trim at all. We can definitely discuss this in a separate PR, though.

Comment (dvoytenko) [inline src/url-parse-query-string.js]: Yeah. I'm leaning toward that as well. Not trimming is better than trimming in this case.
